src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetLockService.cs

```diff
@@ -26,9 +26,11 @@ public sealed class NuGetLockService : INuGetLockService, IDisposable
         {
             if (_lockCount.Value == 0)
             {
-                _lockCount.Value++;
                 await _semaphore.WaitAsync(token);
 
+                // Once this thread acquired the lock then increment lockCount
```

Comment:
It would be good to add a comment explaining the intent behind the entire method.  Something along the lines of:

This method guarantees that only one operation executes at a time globally; however, once an asynchronous call context has acquired the semaphore, reentrancy for that call context is allowed without having to wait again on the semaphore.